tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/CombinationListingFeatureContext.php

```diff
@@ -66,118 +67,193 @@ private function fillCommand(UpdateCombinationFromListingCommand $command, array
+        if (isset($dataRows['is default'])) {
+            $filters[$this->getDbField('is default')] = PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['is default']);
         }
+
+        $limit = isset($dataRows['limit']) ? (int) $dataRows['limit'] : $defaults['limit'];
+        $offset = isset($dataRows['page']) ? $this->countOffset((int) $dataRows['page'], $limit) : $defaults['offset'];
+        $orderBy = isset($dataRows['order by']) ? $this->getDbField($dataRows['order by']) : $defaults['orderBy'];
+        $orderWay = isset($dataRows['order way']) ? $this->getDbField($dataRows['order way']) : $defaults['sortOrder'];
+
+        return new CombinationFilters([
+            'limit' => $limit,
+            'offset' => $offset,
+            'orderBy' => $orderBy,
+            'sortOrder' => $orderWay,
+            'filters' => $filters,
+        ]);
     }
 
     /**
-     * @Then product :productReference should have no combinations
+     * @When I search product ":productReference" combinations list by following search criteria:
      *
      * @param string $productReference
+     * @param CombinationFilters $combinationFilters
      */
-    public function assertNoCombinations(string $productReference): void
+    public function storeSearchCriteria(string $productReference, CombinationFilters $combinationFilters): void
     {
-        $combinationsList = $this->getCombinationsList($productReference);
-
-        Assert::assertEmpty($combinationsList->getCombinations());
-        Assert::assertEquals(0, $combinationsList->getTotalCombinationsCount());
+        $this->getSharedStorage()->set($this->getSearchCriteriaKey($productReference), $combinationFilters);
     }
 
     /**
-     * @param int $page
-     * @param int $limit
+     * @param string $field
      *
-     * @return int
+     * @return string
      */
-    private function countOffset(int $page, int $limit): int
+    private function getDbField(string $field): string
     {
-        return ($page - 1) * $limit;
+        $fieldMap = [
+            'impact on price' => 'price',
+            'is default' => 'default_on',
+        ];
+
+        if (isset($fieldMap[$field])) {
+            return $fieldMap[$field];
+        }
+
+        return $field;
     }
 
     /**
      * @param string $productReference
      * @param array $dataRows
-     * @param int|null $limit
-     * @param int|null $offset
+     * @param bool $wholeList if true then search criteria wont be applied
      */
-    private function assertPaginatedCombinationList(string $productReference, array $dataRows, ?int $limit = null, ?int $offset = null): void
+    private function assertCombinations(string $productReference, array $dataRows, bool $wholeList = false): void
     {
-        $combinationsList = $this->getCombinationsList($productReference, $limit, $offset);
+        $searchCriteriaKey = $this->getSearchCriteriaKey($productReference);
+        if ($wholeList) {
```

Comment:
Instead of creating an empty filter object, I'd allow `getCombinationsList` to have a nullable `$combinationFilters` This it doesn't set the parameters when it is null

Reply:
yeah, been there, for some reason I found thi scleaner, but it makes sense, Ill bring it back